enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/compiled/codegen/ir/WhileLoop.scala

```diff
@@ -33,6 +33,7 @@ case class WhileLoop(variable: Variable, producer: LoopDataGenerator, action: In
         producer.getNext(variable, iterator, loopBody)
         action.body(loopBody)
       }
+      producer.close(iterator, generator)
```

Comment:
Does the generated code also close the cursor if the loop body throws an exception?

Reply:
yes, I had thought of all exits except the happy path 🤔